### ISSUE TITLE
RAM usage

### ISSUE THREAD

**Issue by HostOnNet** [HikariKnight/rsu-client#98]
My RAM usage (VSIZE) is 5.2 GB

> VmPeak:	 5525716 kB

```
boby@hon-pc-01:/proc/8668 $ cat status
Name:	java
State:	S (sleeping)
Tgid:	8668
Ngid:	0
Pid:	8668
PPid:	8667
TracerPid:	0
Uid:	1000	1000	1000	1000
Gid:	1000	1000	1000	1000
FDSize:	128
Groups:	4 24 27 30 46 108 124 129 130 1000 
VmPeak:	 5525716 kB
VmSize:	 5400788 kB
VmLck:	       0 kB
VmPin:	       0 kB
VmHWM:	  860092 kB
VmRSS:	  859552 kB
VmData:	 5036512 kB
VmStk:	     136 kB
VmExe:	       4 kB
VmLib:	   57796 kB
VmPTE:	    2456 kB
VmSwap:	       0 kB
Threads:	42
SigQ:	0/47506
SigPnd:	0000000000000000
ShdPnd:	0000000000000000
SigBlk:	0000000000000000
SigIgn:	0000000000000000
SigCgt:	2000000181005ccf
CapInh:	0000000000000000
CapPrm:	0000000000000000
CapEff:	0000000000000000
CapBnd:	0000001fffffffff
Seccomp:	0
Cpus_allowed:	ffff
Cpus_allowed_list:	0-15
Mems_allowed:	00000000,00000001
Mems_allowed_list:	0
voluntary_ctxt_switches:	1
nonvoluntary_ctxt_switches:	3
boby@hon-pc-01:/proc/8668 $ 
```

I have total of  6G RAM, so this cause some swap usage.

Anyway i can limit the RAM  usage by RuneScape ?

I know this is not a bug or issue related to rsu, but any help ?

**Comment by ivanpu** [comment]
Yes, in RSU Go into `Settings` > `Client Settings`, and change `Maximum Java Heap Allocation` to whatever you want.

**Comment by HostOnNet** [comment]
Can i set it editing any files like settings.conf ?

I don't have the WX perl installed. Not sure if i need it installed just for this..

Just done little googling and found Java Heap Memory is set with -Xmx parm. The command run to start game by rsu have it set to 512 MB (-Xmx512m)

```
/usr/lib/jvm/java-8-oracle/jre/bin/java -XX:+AggressiveOpts -XX:+UnlockExperimentalVMOptions -XX:+DisableExplicitGC -XX:+TieredCompilation -XX:ReservedCodeCacheSize=256m -XX:+UseAdaptiveGCBoundary -XX:SurvivorRatio=16 -XX:+UseParallelGC -Dhttps.protocols=TLSv1,TLSv1.1,TLSv1.2 -Duser.home=/home/boby -cp jagexappletviewer.jar -Dcom.jagex.config=http://www.runescape.com/k=3/l=0/jav_config.ws -Xmx512m -Xss2m -Dsun.java2d.noddraw=true jagexappletviewer /share/img/RuneScape3
```

You can see 5.6 G usage in atop

![atop](https://cloud.githubusercontent.com/assets/78068/15175820/d5f39602-1785-11e6-80d1-92326df48b93.png)

**Comment by ivanpu** [comment]
Try adding `-Xms512m` (or anything else, less then `-Xmx`) parameter to `runescape.prm`, to force smaller initial heap.

I'm not aware of anything else that might cause this growth, unless there is memory leak inside Java itself, or there is another java process running for something else and it's the one you see on the top of the top.

**Comment by HostOnNet** [comment]
Which version of java everyone else using ?

I am using Java version

```
boby@hon-pc-01:~ $ java -version
java version "1.8.0_31"
Java(TM) SE Runtime Environment (build 1.8.0_31-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.31-b07, mixed mode)
boby@hon-pc-01:~ $ 
```

Should i use a different version ?

OpenJRE available in Ubuntu 14.04 is  "openjdk-7-jre",  should i try this ?

**Comment by HostOnNet** [comment]
Change java version to latest sun java

```
boby@hon-pc-01:~ $ java -version
java version "1.8.0_92"
Java(TM) SE Runtime Environment (build 1.8.0_92-b14)
Java HotSpot(TM) 64-Bit Server VM (build 25.92-b14, mixed mode)
boby@hon-pc-01:~ $ 
```
But still take same amount of virtal memory.

```
boby@hon-pc-01:~ $ cat /proc/11444/status | grep -i vmsize
VmSize:  5468488 kB
boby@hon-pc-01:~ $ 
```

This is with 

```
-Xms512m -Xmx512m
```

**Comment by ivanpu** [comment]
> Which version of java everyone else using ?

```
java version "1.7.0_99"
OpenJDK Runtime Environment (IcedTea 2.6.5) (Arch Linux build 7.u99_2.6.5-1-x86_64)
OpenJDK 64-Bit Server VM (build 24.95-b01, mixed mode)
```

**Comment by HostOnNet** [comment]
Tested openjdk-7-jre, not a big differnce in VmSize, it use 1 GB less than Oracle Java.

With latest Sun Java 

```
$ cat /proc/5624/status | grep -i vmsize
VmSize:  5607076 kB
$ 
```

With  openjdk-7-jre (available on Ubuntu 14.04 repo)

```
$ java -version
java version "1.7.0_101"
OpenJDK Runtime Environment (IcedTea 2.6.6) (7u101-2.6.6-0ubuntu0.14.04.1)
OpenJDK 64-Bit Server VM (build 24.95-b01, mixed mode)
$ cat /proc/10680/status | grep -i vmsize
VmSize:  4463200 kB
$ 
```
